app/javascript/components/reconfigure-vm-form/helpers/disk.js

```diff
@@ -0,0 +1,351 @@
+  const index = id.substring(4);
+  data.dataTable.disks[index].hdSize = data.dataTable.disks[index].old_size;
+  data.dataTable.disks[index].hdUnit = data.dataTable.disks[index].old_unit;
+  const submitDisks = data.submitParams.disks.resize.filter((disk) => disk.id !== row.id);
+  const resizeDiskIds = data.submitParams.diskIds.resize.filter((id) => id !== row.id);
+  data.changeCounter -= 1;
+  setData({
+    ...data,
+    submitParams: {
+      ...data.submitParams,
+      disks: {
+        ...data.submitParams.disks,
+        resize: submitDisks,
+      },
+      diskIds: {
+        ...data.submitParams.diskIds,
+        resize: resizeDiskIds,
+      },
+    },
+  });
+};
+
+/** Function to check the button need to be disabled or not */
+const diskDisableButton = (id, data) => {
+  if (data.submitParams.diskIds.delete.includes(id) || data.submitParams.diskIds.add.includes(id)) {
+    return true;
+  }
+  return false;
+};
+
+/** Function to check the delete button rendering */
+const diskAction = (id, data) => {
+  if (data.submitParams.diskIds.delete.includes(id)) {
+    return {
+      is_button: true,
+      title: __('Cancel Delete'),
+      text: __('Cancel Delete'),
+      alt: __('Cancel Delete'),
+      kind: 'danger',
+      callback: 'cancelDeleteDisk',
+    };
+  }
+  return {
+    is_button: true,
+    title: __('Delete'),
+    text: __('Delete'),
+    alt: __('Delete'),
+    kind: 'danger',
+    disabled: data.submitParams.diskIds.resize.includes(id),
+    callback: 'deleteDisk',
+  };
+};
+
+/** Function to check the resize button rendering */
+const resizeAction = (id, data) => {
+  if (data.submitParams.diskIds.resize.includes(id)) {
+    return {
+      is_button: true,
+      title: __('Cancel Resize'),
+      text: __('Cancel Resize'),
+      alt: __('Cancel Resize'),
+      kind: 'secondary',
+      callback: 'cancelResizeDisk',
+    };
+  }
+  return {
+    is_button: true,
+    title: __('Resize'),
+    text: __('Resize'),
+    alt: __('Resize'),
+    kind: 'secondary',
+    callback: 'resizeDisk',
+    disabled: diskDisableButton(id, data),
+  };
+};
+
+/** Function to validate the disk size
+ * add disk form, resize form uses this same function
+ */
+export const validateDiskSize = (value, unit, minSize, minUnit) => {
+  const valueInMb = sizeInMB(value, unit);
+  const minSizeInMb = sizeInMB(minSize, minUnit);
+  if (valueInMb < minSizeInMb) {
+    return false;
+  }
+
+  return true;
+};
+
+/** Function to set the Disk datatable row data - one time call
+ * This formatting is needed for resize function validation
+ */
+export const formatDiskData = (disks) => {
+  disks.forEach((disk) => {
+    disk.orgHdSize = disk.hdSize;
+    disk.orgHdUnit = disk.hdUnit;
+  });
+
+  return disks;
+};
+
+/** Function to set the Disk datatable row data  */
+export const setDiskData = (disks, roles, data, setData) => {
+  const diskData = [];
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < disks.length; i++) {
```

Comment:
lets try to break the for loop and use forEach if possible

Reply:
done.